crates/ruff_python_parser/src/lexer.rs

```diff
@@ -82,7 +82,7 @@ impl<'src> Lexer<'src> {
     /// If the start offset is greater than 0, the cursor is moved ahead that many bytes.
     /// This means that the input source should be the complete source code and not the
     /// sliced version.
-    pub(crate) fn new(source: &'src str, mode: Mode, start_offset: TextSize) -> Self {
+    pub fn new(source: &'src str, mode: Mode, start_offset: TextSize) -> Self {
```

Comment:
I'd be okay to have a function next to `lex_tokens` that also takes an offset but I rather keep the constructor `pub(crate)`

Reply:
np.But it does feel a bit redundant as it would have the same signature and would call `Lexer::new`, so we end up with two identical methods that are the same.

I'm not even sure how to call it :sweat_smile: